compair/models/comparison.py

```diff
@@ -142,12 +143,56 @@ def _get_new_comparison_pair(cls, course_id, assignment_id, user_id, pairing_alg
 
         comparison_pairs = [comparison.convert_to_comparison_pair() for comparison in comparisons]
 
-        comparison_pair = generate_pair(
-            package_name=pairing_algorithm.value,
-            scored_objects=scored_objects,
-            comparison_pairs=comparison_pairs,
-            log=current_app.logger
-        )
+        # adaptive min delta algo requires extra criterion specific parameters
+        if pairing_algorithm == PairingAlgorithm.adaptive_min_delta:
+            # retreive extra criterion score data
+            answers_with_criterion_score = Answer.query \
```

Comment:
you can optionally simplify this

```
            answers_with_criterion_score = Answer.query \
                .with_entities(Answer, AnswerCriterionScore.criterion_id, AnswerCriterionScore.score) \
                .join(AnswerCriterionScore) \
                .filter(and_(
                    Answer.user_id.notin_(ineligible_user_ids),
                    Answer.assignment_id == assignment_id,
                    Answer.active == True,
                    Answer.practice == False,
                    Answer.draft == False
                )) \
                .all()
```

since your not really using the Answer part of the query (`answer_criterion_score` has an `answer_id` for filling in the hash)

Reply:
revised.  pls review the commit before i proceed with squashing